openpype/hosts/nuke/plugins/publish/collect_writes.py

```diff
@@ -48,113 +27,95 @@ def process(self, instance):
-                    ).format(first_frame - 1)
-
-                    slate_frame = collected_frames[0].replace(
-                        frame_start_str, frame_slate_str)
-                    collected_frames.insert(0, slate_frame)
-
-                if collected_frames_len == 1:
-                    representation['files'] = collected_frames.pop()
-                else:
-                    representation['files'] = collected_frames
-
-                # inject colorspace data
-                self.set_representation_colorspace(
-                    representation, instance.context,
-                    colorspace=colorspace
-                )
+            self.set_farm_representation(
+                instance,
+                write_file_path,
+                frame_start,
+                frame_end,
+                only_existing=True
+            )
+
+        elif render_target == "frames_farm":
+            self.set_farm_representation(
+                instance,
+                write_file_path,
+                frame_start,
+                frame_end,
+                only_existing=True
+            )
 
-            instance.data["representations"].append(representation)
-            self.log.info("Publishing rendered frames ...")
+            self.add_farm_instance_data(instance)
 
         elif render_target == "farm":
-            farm_keys = ["farm_chunk", "farm_priority", "farm_concurrency"]
-            for key in farm_keys:
-                # Skip if key is not in creator attributes
-                if key not in creator_attributes:
-                    continue
-                # Add farm attributes to instance
-                instance.data[key] = creator_attributes[key]
-
-            # Farm rendering
-            instance.data["transfer"] = False
-            instance.data["farm"] = True
-            self.log.info("Farm rendering ON ...")
+            self.add_farm_instance_data(instance)
+
+        # set additional instance data
+        self._set_additional_instance_data(
+            instance, write_node, frame_start, frame_end, colorspace)
+
+    def _get_frame_range(self, write_node):
+        """Get frame range data from instance.
+
+        Args:
+            write_node (nuke.Node): write node
+
+        Returns:
+            tuple: frame_start, frame_end
+        """
+        # Get frame range from workfile
+        frame_start = int(nuke.root()["first_frame"].getValue())
+        frame_end = int(nuke.root()["last_frame"].getValue())
+
+        # Get frame range from write node if activated
+        if write_node["use_limit"].getValue():
+            frame_start = int(write_node["first"].getValue())
+            frame_end = int(write_node["last"].getValue())
+
+        return frame_start, frame_end
+
+    def _set_additional_instance_data(
+        self, instance, write_node, frame_start, frame_end, colorspace
+    ):
+        """Set additional instance data.
+
```

Comment:
It would be great if the docstring could elaborate on what "additional data" is. What is it setting?